ts/sagas/wallet.ts

```diff
@@ -24,10 +90,203 @@ function* fetchTransactions(
 }
 
 function* fetchCreditCards(
-  loadCards: () => Promise<ReadonlyArray<CreditCard>>
+  loadCards: () => Promise<ReadonlyArray<Wallet>>
+): Iterator<Effect> {
+  const cards: ReadonlyArray<Wallet> = yield call(loadCards);
+  yield put(walletsFetched(cards));
+}
+
+const navigateTo = (routeName: string, params?: object) => {
+  return NavigationActions.navigate({ routeName, params });
+};
+
+function* paymentSagaFromQrCode(): Iterator<Effect> {
+  yield put(paymentQrCode());
+  yield put(navigateTo(ROUTES.PAYMENT_SCAN_QR_CODE)); // start by showing qr code scanner
+  yield fork(paymentSaga);
+}
+
+function* paymentSaga(): Iterator<Effect> {
+  while (true) {
+    const action = yield take([
```

Comment:
I am quite concerned that there is no enforcement of a certain sequence of actions in this saga... 

For instance, `PAYMENT_REQUEST_CONFIRM_PAYMENT_METHOD` may come only after a `PAYMENT_REQUEST_PICK_PAYMENT_METHOD`.

@fpersico do you have any suggestion on how to handle this flow?

Reply:
Enforcing a correct sequence of actions is non-trivial: consider that the attached graph is a simplified version that does not include the "back", "cancel" and "tap on banner" actions that can occur in any given screen, so the number of edges in that graph almost triples. 

A *custom* solution would have to define a directed graph with the available nodes and the allowed (directed) edges plus, for each edge, a "converting" function that maps the origin state to the destination one (e.g. when going from the "pick payment method" to the "confirm payment method", the `selectedPaymentMethod` value should be added to the state, while going back from "confirm payment method" to the "payment summary" (by tapping the payment banner on the top of the screen) would have to clear the stored information (psp, selected wallet). 

This is not to say that it would be infeasible, but it would imply a significant amount of work and it might make sense to have this done at a later time. 

Even if a library that simplifies this kind of work exists, I would still expect it to require that those nodes and edges be defined.